docs/kit/hardware/kch.md

```diff
@@ -0,0 +1,86 @@
+# KCH (Raspberry Pi HAT)
+
+![3D Render of KCH](img/kch-3d-render.png)
+
+## KCH LEDs
+
+The LEDs on the KCH are split into several groups, some indicate the power state of the KCH, some indicate the state of the robot and some are directly controlled by student code.
+
+### Power Supply
+
+- `5V` (Green)
+    - 5V power is present, whether from the USB-C port on the Pi or the KCH regulator.
+- `Reg` (Green)
+    - The regulator on the KCH is enabled.
+- `Power in` (Green)
+    - 12V power is present on the 12V input.
+- `Reverse polarity` (Red)
+    - The 12V power input has reversed polarity.
+
+### Boot Progress LEDs
+
+There are 5 green LEDs that are a "progress bar" of boot progress on the KCH. Each represents a different stage of boot:
+
+- 20%
+    - Connected to GPIO 7, which is configured as [SPI0 CE1 (ALT0)](https://pinout.xyz/pinout/pin26_gpio7)
+    - It is configured in the [KCH EEPROM](https://github.com/srobo/kch-hw/blob/main/EEPROM/eeprom_settings.txt#L69).
+    - In normal behaviour, it turns on when the GPIO map is loaded from the EEPROM, i.e in the videocore bootloader (which requires a start*.elf).
+    - This LED indicates that the SD card can be read and that the operating system is loading.
+    - This LED will turn on with any valid Raspberry Pi distro.
+- 40%
+    - This LED is turned on by [`boot_40.service`](https://github.com/srobo/robot-image/blob/main/sources/meta-srobo/recipes-robot/python-astoria/astoria-config/boot_40.service), which runs in the early stages of systemd (`basic.target`).
+    - This LED indicates that the filesystem on the SD card is valid.
+    - It also indicates that the SD card contains a Student Robotics OS image.
+- 60%
+    - This LED is controlled by [`kchd`](../../software/#kch-daemon-kchd) and indicates `kchd` is running.
+    - If this LED has turned on, most of the operating system is loaded and behaving correctly.
+- 80%
+    - This LED is turned on when the MQTT Event Broker is running
+- 100%
+    - This LED is turned on when all critical astoria services are loaded and running.
+    - It will turn off if any of the services stop or crash.
+
+### Status LEDs
+
+The following LEDs are driven by GPIO on the Raspberry Pi and indicate the status of the robot.
+
+All except the `Start` and Heartbeat LED are controlled by [`kchd`](../../software/#kch-daemon-kchd). If `kchd` is not running, these LEDs will not light up. Check the boot progress LEDs to confirm.
+
+- `Code` (Green)
+    - Enabled when a USB drive containing a usercode file (`robot.py`) is mounted.
```

Comment:
It will still light up. It is on when any drive containing a `robot.py` is mounted and known by Astoria.

> something critical like a syntax error

This is an error that occurs in the Python process, so would be indicated by the `OK` LED turning red. This is because a syntax error results in a negative exit code.